### DISCUSSION TITLE
Migrating Tekton images from gcr.io to ghcr.io

### DISCUSSION THREAD

**Discussion by afrittoli** [tektoncd/community#1112] [General]
Tekton releases and nightly builds are backed today by storage services provided by Google Cloud.
The release YAMLs are stored in buckets (object storage), while OCI images are stored in the `gcr.io` registry, in four different zones, along with image signatures and SBOMs.

Tekton users as well as upstream and downstream CI systems download Tekton images on a daily basis. It's great to see so many downloads of Tekton's container images. The downside of this, however, is that the cloud egress traffic represents today more than 50% of the total Tekton CI/CD cloud costs. 

Apart from concerns related to cloud costs, the Google container registry `gcr.io` is now [deprecated in favour of the new artifact registry](https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr), hosted at `pkg.dev`, and that migration would require some work in any case.

GitHub packages are [free of charge for public repositories](https://github.com/features/packages), so I would like to propose migrating our release pipelines from `gcr.io` to `ghcr.io`. Initial steps could be:
- the OCI images of future nightly builds will be stored on `ghcr.io`. The yaml will continue to be hosted, for the time being, on `gcr.io`
- the OCI images of future releases will be store will be stored on `ghcr.io`. The yaml will continue to be hosted, for the time being, on both `gcr.io` and GitHub

Over time we could migrate LTS releases to `ghcr.io` and use a proxy to continue supporting `gcr.io`.
This should be enough to significantly reduce the egress traffic. Eventually we could migrate the data for EOL releases and old nightly builds as well.

**Comment by chmouel** [comment]
We would maybe loose some  insights statistics isn't it ? I think GHCR provides only the number of downloads with a small graph:

![image](https://github.com/tektoncd/community/assets/98980/5bc3a125-9a11-4cf7-a72e-e6c3f55116c7)

just maybe something to be aware, aside of that i think moving GHCR make sense
